x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/logging/DeprecationIndexingComponent.java

```diff
@@ -74,17 +86,59 @@ public DeprecationIndexingComponent(
             consumer
         );
         enableDeprecationLogIndexing(enableDeprecationLogIndexingDefault);
+
+    }
+
+    public static DeprecationIndexingComponent createDeprecationIndexingComponent(
+        Client client,
+        Settings settings,
+        RateLimitingFilter rateLimitingFilterForIndexing,
+        boolean enableDeprecationLogIndexingDefault,
+        ClusterService clusterService
+    ) {
+        final DeprecationIndexingComponent deprecationIndexingComponent = new DeprecationIndexingComponent(
+            client,
+            settings,
+            rateLimitingFilterForIndexing,
+            enableDeprecationLogIndexingDefault,
+            clusterService
+        );
+
+        clusterService.addListener(deprecationIndexingComponent);
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(WRITE_DEPRECATION_LOGS_TO_INDEX, deprecationIndexingComponent::enableDeprecationLogIndexing);
+
+        return deprecationIndexingComponent;
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (flushEnabled.get() == false) {
```

Comment:
Maybe instead of wrapping multiple if statements, maybe do this:

```java
if (flushEnabled.get()) {
   return;
}

if (event.metadataChanged() == false) {
   return;
}

// rest of the code
```

I think this is more readable.